src/Components/Header/SearchBar.jsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useRef, useState } from "react";
+
+const searchKeywords = [
+  "Brustro",
+  "Bianyo",
+  "Camel",
+  "DOMS",
+  "Faber Castell",
+  "Staedtler",
+  "Blenders",
+  "Colours",
+  "Colour Palettes",
+  "Drawing Boards",
+  "Drawing Pencils",
+  "Drawing Surfaces",
+  "Eraser",
+  "Portfolio Storage",
+  "Painting Brush",
+  "Painting Knives"
+];
+
+export const SearchBar = () => {
+  const [activeSearchIcon, setSearchIconActive] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const searchBarRef = useRef(null);
+  const closeSearchBar = (e) => {
+    if (!searchBarRef.current.contains(e.target)) {
+      setSearchIconActive(false);
+    }
+  };
+  useEffect(() => {
+    document.addEventListener("click", closeSearchBar);
+
+    return () => {
+      document.removeEventListener("click", closeSearchBar);
+    };
+  }, []);
+

```

Comment:
I am not sure of this here, why do you require vanilla js to close searchbar ?

Reply:
I m trying to add on click on anything in the document except for the search bar should close the search bar. 
I found this way, if there is any other way to do it in react then please let me know.